### DISCUSSION TITLE
Editieren - AutoComplete

### DISCUSSION THREAD

**Discussion by Andreas-Lampert** [e-netze/webgis-community#46] [General]
Hallo Jürgen!

Tritt bei mir mit Version 5.23.901 auf
Ich habe im CMS beim Editieren das Attribut Gewässer mittels Eingabe Typ "AutoComplete" konfiguriert (siehe Screenshot). 

![autocomplete_edit_cms](https://user-images.githubusercontent.com/122986367/223406931-4be0da1e-5f73-4efc-9eeb-33af1b6b3339.png)

Bei der Eingabe im Feld Gewässer in der Editier-Maske erscheint allerdings keine Werteliste und in der Entwicklerkonsole wird ein Fehler ausgegeben (siehe Screenshot). 

![autocomplete_edit_error](https://user-images.githubusercontent.com/122986367/223406938-96852297-29d3-4107-8732-c35a88d9fe18.png)

Die Konfiguration der Felder im CMS im Bereich "Database Domain" ist analog zum webgis4. Dort funktioniert es einwandfrei. Muss ich in der Konfiguration im webgis5 irgendwas ändern/anpassen, um den Eingabe Typ "AutoComplete" wieder verfügbar zu haben?

Danke!

Lg, Andi

**Comment by gis-eni** [comment]
Anscheinend liefert deine Auswahlliste kein gültiges Array mit Werten zurück. Dann kann diese Fehlermeldung zustande kommen.
Was vielleicht noch interessant wäre:

* Öffne in den Entwickertools im Browser einmal den Breich "network". Dort kannst du auch noch nach "Fetch/XHR" filtern.
* Wenn du jetzt etwas in dein AutoComplete Textfeld eingibst, sollte bei jeden Tastendruck ein Request an den Server gehen, der die Werte abholt. Was ist denn bei so einem Request der Response? Es sollte ein Json zurück kommen.

**Comment by Andreas-Lampert** [comment]
Danke für die Info und die Tipps!

Das Problem war die Where-Klausel. Die muss befüllt werden. 

Im Preview gibt es folgenden Fehler:

![where_incorrect](https://user-images.githubusercontent.com/122986367/223461843-ae1377d9-45d6-44ba-af56-f6607b1e8b16.png)

Nach der Konfiguration im CMS mit einer Where-Klausel geht die AutoComplete-Liste.

![where_cms](https://user-images.githubusercontent.com/122986367/223461399-28cbca20-549e-4ce4-8bcc-d221f11b258e.png)

Eintrag kann geschlossen werden. 


Lg, Andi

**Comment by gis-eni** [comment]
Stimmt, hab ich auch nicht gesehen. Manchmal ist es so einfach ;)

LG Jürgen
